cfgrammar/src/lib/yacc/ast.rs

```diff
@@ -25,10 +25,10 @@ impl ASTWithValidityInfo {
     /// encountered during the construction of it.  The `ASTWithValidityInfo` can be
     /// then unused to construct a `YaccGrammar`, which will either produce an
     /// `Ok(YaccGrammar)` or an `Err` which includes these errors.
-    pub fn new(yacc_kind_resolver: YaccKindResolver, s: &str) -> Self {
+    pub fn new(header: Header, s: &str) -> Self {
```

Comment:
So, I wonder here if we can take a `header: &mut Header` here,
and then avoid returning it in `YaccParser::build()`, it seems possible that would simplify things.

But I'm not sure whether the compiler will be able to tell that it isn't capture by the `GrammarAST` return value.

Reply:
It appears that we can, this was done in [4c59bb9](https://github.com/softdevteam/grmtools/pull/547/commits/4c59bb93225d6dda2b0a93bdca2fcdf9734e32c1).

It required adding a lifetime to (now) `YaccParser<'a>`, but I think that should be a private type.
Seemed like a small win to me.